### PR TITLE
[PHP 5.5] [RFC] Future version bump to PHP 5.5 and Symfony 2.7

### PR DESCRIPTION
**Pull request by pamil**
Sylius requirements will bump in near future, so it's initial PR to do it. Will be helpful for Sylius developers, who code with the latest dependencies in mind.

### CODE REVIEW

**Comment by TomasVotruba** [comment]
:+1:

**Comment by pamil** [comment]
You're free to open a new PR based on mine, this one should include only necessary changes to made the version bump. There are #3055 and #2844 PRs just waiting for bump.

I don't think it's official yet, but the security support for PHP5.4 itself ends in a month, so I would find it as a rational decision.

@pjedrzejewski @michalmarcinkowski what do you think?

**Comment by TomasVotruba** [comment]
Thank you for answer. I shall send PR when this get merged.

I know many maintainers that keep PHP 5.3 support from fear of dropping package users, so it's not an easy step. Personally I create new packages with PHP 5.6 as minimum.

File: composer.json [hunk 1]
```diff
@@ -19,16 +19,16 @@
         }
     ],
     "require": {
-        "php":                                  ">=5.3.9",
+        "php":                                  "~5.5",
```
Comment (pjedrzejewski) [inline composer.json]: @pamil Should be >=5.5.9.

Comment (pamil) [inline composer.json]: Done :)

**Comment by pjedrzejewski** [comment]
@pamil Requires rebasing. :)

File: composer.json [hunk 2]
```diff
@@ -19,15 +19,15 @@
...
         "a2lix/translation-form-bundle":        "~2.0",
         "doctrine/collections":                 "~1.2",
-        "doctrine/common":                      "~2.3",
+        "doctrine/common":                      "~2.5",
         "doctrine/doctrine-bundle":             "~1.3",
         "doctrine/doctrine-cache-bundle":       "~1.0",
         "doctrine/doctrine-fixtures-bundle":    "~2.2",
         "doctrine/doctrine-migrations-bundle":  "~1.0",
-        "doctrine/orm":                         "~2.4",
+        "doctrine/orm":                         ">=2.4.8,<2.5",
```
Comment (stloyd) [inline composer.json]: Any reference for this lock?

Comment (pamil) [inline composer.json]: Version `2.5.0` has security issue and version `2.5.1` fixes that issue but introduces some bugs that affects our `@products` suite.

Comment (pamil) [inline composer.json]: And versions lower than `2.4.8` have that security issue too :)

**Comment by pamil** [comment]
@pjedrzejewski updated and rebased, waiting for green light :)

**Comment by pjedrzejewski** [comment]
Thank you Kamil! :+1:

**Comment by michalmarcinkowski** [comment]
@pamil we also need a note about that in a changelog

**Comment by pjedrzejewski** [comment]
@pamil Please add it to changelog. :)

**Pull request merged**